haystack/nodes/answer_generator/openai.py

```diff
@@ -314,7 +254,7 @@ def _build_prompt_within_max_length(self, query: str, documents: List[Document])
         skipped_docs = 0
         # If leftover_token_len is negative we have gone past the MAX_TOKENS_LIMIT and the prompt must be trimmed
         if leftover_token_len < 0:
-            n_docs_tokens = [self._count_tokens(doc.content) for doc in documents]
+            n_docs_tokens = [count_openai_tokens(text=doc.content, tokenizer=self._tokenizer) for doc in documents]
```

Comment:
Yeah, I think this is a good idea. This would avoid a user accidentally sending 100+ docs to the PromptNode or AnswerGenerator and wonder what is taking so long. But this PR is already fairly large so I think opening a separate issue for this would be a good idea.